great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -2634,441 +2632,12 @@ def test_yaml_config(
-            store_name=store_name, store_obj=instantiated_class
-        )
-        return instantiated_class, usage_stats_event_payload
-
-    def _test_instantiation_of_datasource_from_yaml_config(
-        self, name: Optional[str], class_name: str, config: CommentedMap
-    ) -> Tuple[Datasource, dict]:
-        """
-        Helper to create datasource instance and update usage stats payload.
-        See `test_yaml_config` for more details.
-        """
-        print(f"\tInstantiating as a Datasource, since class_name is {class_name}")
-        datasource_name: str = name or config.get("name") or "my_temp_datasource"
-        instantiated_class = cast(
-            Datasource,
-            self._instantiate_datasource_from_config_and_update_project_config(
-                name=datasource_name,
-                config=config,
-                initialize=True,
-            ),
-        )
-
-        anonymizer = Anonymizer(self.data_context_id)
-
-        if class_name == "SimpleSqlalchemyDatasource":
-            # Use the raw config here, defaults will be added in the anonymizer
-            usage_stats_event_payload = anonymizer.anonymize(
-                obj=instantiated_class, name=datasource_name, config=config
-            )
-        else:
-            # Roundtrip through schema validation to remove any illegal fields add/or restore any missing fields.
-            datasource_config = datasourceConfigSchema.load(instantiated_class.config)
-            full_datasource_config = datasourceConfigSchema.dump(datasource_config)
-            usage_stats_event_payload = anonymizer.anonymize(
-                obj=instantiated_class,
-                name=datasource_name,
-                config=full_datasource_config,
-            )
-        return instantiated_class, usage_stats_event_payload
-
-    def _test_instantiation_of_checkpoint_from_yaml_config(
-        self, name: Optional[str], class_name: str, config: CommentedMap
-    ) -> Tuple[Checkpoint, dict]:
-        """
-        Helper to create checkpoint instance and update usage stats payload.
-        See `test_yaml_config` for more details.
-        """
-        print(f"\tInstantiating as a {class_name}, since class_name is {class_name}")
-
-        checkpoint_name: str = name or config.get("name") or "my_temp_checkpoint"
-
-        checkpoint_config: Union[CheckpointConfig, dict]
-
-        checkpoint_config = CheckpointConfig.from_commented_map(commented_map=config)
-        checkpoint_config = checkpoint_config.to_json_dict()
-        checkpoint_config.update({"name": checkpoint_name})
-
-        checkpoint_class_args: dict = filter_properties_dict(
-            properties=checkpoint_config,
-            delete_fields={"class_name", "module_name"},
-            clean_falsy=True,
-        )
-
-        if class_name == "Checkpoint":
-            instantiated_class = Checkpoint(data_context=self, **checkpoint_class_args)
-        elif class_name == "SimpleCheckpoint":
```

Comment:
[non-blocking]
I'm missing a lot of context here so please take this with a grain of salt but...

Do you know why we are doing this string comparison on the class name rather than an `isinstance` or `issubclass` check?

Reply:
Good question! At this point we are doing a comparison with the string class name from the yaml config passed in. We don't actually have an instantiated class yet.